infra/ci/worker/sandbox_runner.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+# Copyright (C) 2025 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+''' Runs one GitHub (Ephemeral) Action Runner and quits at the end.
+
+The Action Runner is executed in its own sandboxed docker container, which has
+no access to the metadata server.
+It also handles graceful container termination upon shutdown.
+'''
+
+import logging
+import os
+import signal
+import socket
+import subprocess
+import time
+import sys
+
+from config import SANDBOX_IMG, GITHUB_REPO, SANDBOX_SVC_ACCOUNT
+from common_utils import get_github_registration_token
+
+CUR_DIR = os.path.dirname(__file__)
+
+# The container name will be GCE_HostName-N.
+SANDBOX_ID = os.environ.get('SANDBOX_ID', '')
+SANDBOX_NAME = '%s-%s' % (socket.gethostname(), SANDBOX_ID)
+
+
+def sig_handler(_, __):
+  logging.warning('Interrupted by signal, exiting worker')
+  subprocess.call(['docker', 'stop', SANDBOX_NAME])
+  sys.exit(0)
+
+
+def create_sandbox_token(token_path):
+  # Impersonate the sandbox service account. This creates a short-lived 1h token
+  # for a downgraded service account that we pass to the sandbox. The sandbox
+  # service account is allowed only storage object creation for untrusted CI
+  # artifacts.
+  sandbox_svc_token = subprocess.check_output([
+      'gcloud',
+      'auth',
+      'application-default',
+      'print-access-token',
+      '--impersonate-service-account=%s' % SANDBOX_SVC_ACCOUNT,
+      # TODO(primiano): uncomment below after b/406184705 is fixed.
```

Comment:
This is fixed.

Reply:
Unfortunately seems to have not propagated yet:

root@gh-us-west1-7svl:/opt/ci# gcloud auth application-default print-access-token --impersonate-service-account=gce-ci-sandbox@perfetto-ci.iam.gserviceaccount.com --lifetime=21600
ERROR: (gcloud.auth.application-default.print-access-token) There was a problem refreshing your current auth tokens: ('Unable to acquire impersonated credentials', '{\n  "error": {\n    "code": 400,\n    "message": "Access tokens for service account gce-ci-sandbox@perfetto-ci.iam.gserviceaccount.com can not have lifetimes of more than 1 hour. Use the Org Policy Constraint constraints/iam.allowServiceAccountCredentialLifetimeExtension to allow up to 12 hours.",\n    "status": "INVALID_ARGUMENT"\n  }\n}\n')